crates/miden-tx/src/executor/mod.rs

```diff
@@ -360,31 +360,41 @@ where
         .map_err(TransactionExecutorError::TransactionProgramExecutionFailed);
 
         match result {
-            Ok(_) => Ok(NoteAccountExecution::Success),
-            Err(tx_execution_error) => {
+            Ok(_) => Ok(NoteConsumptionInfo::new_successful(
+                tx_inputs
+                    .into_input_notes()
+                    .into_iter()
+                    .map(|note| note.into_note())
+                    .collect::<Vec<_>>(),
+            )),
+            Err(error) => {
+                // Map the last note id from execution to the failed note.
                 let notes = host.tx_progress().note_execution();
-
-                // empty notes vector means that we didn't process the notes, so an error
-                // occurred somewhere else
-                if notes.is_empty() {
-                    return Err(tx_execution_error);
-                }
-
-                let ((last_note, last_note_interval), success_notes) = notes
+                let ((last_note, _last_note_interval), success_notes) = notes
                     .split_last()
                     .expect("notes vector should not be empty because we just checked");
-
-                // if the interval end of the last note is specified, then an error occurred after
-                // notes processing
-                if last_note_interval.end().is_some() {
-                    return Err(tx_execution_error);
-                }
-
-                Ok(NoteAccountExecution::Failure {
-                    failed_note_id: *last_note,
-                    successful_notes: success_notes.iter().map(|(note, _)| *note).collect(),
-                    error: Some(tx_execution_error),
-                })
+                let note = input_notes
+                    .iter()
+                    .find(|&note| note.id() == *last_note)
+                    .expect("Last note returned from note execution should exist in input notes")
+                    .note()
+                    .clone();
```

Comment:
I think with the removal of the `notes.is_empty()` check this first `expect`ation no longer holds.